### ISSUE TITLE
Explain issues in more detail

### ISSUE THREAD

**Issue by kuza55** [Manishearth/rust-clippy#105]
This is supposedly targeted at newbies, but it's really not clear why some of these are prefered.

E.g. I'm pretty confused why uses of LinkedList should be replaced by RingBuf.

It would be great if there was a wiki to which the output linked to.

**Comment by Manishearth** [comment]
Sounds good. @Gankro want to start?

**Comment by Gankro** [comment]
@Manishearth I'm honestly pretty swamped writing *other* documentation. :/

The TL;DR of LinkedList is that it's built on a massive amount of pointers and indirection. It wastes memory, it has terrible cache locality, and is all-around slow. RingBuf, while "only" amortized for push/pop, should be faster in the general case for *almost* every possible workload, and isn't even amortized at all if you can predict the capacity you need.

LinkedLists are only *really* good if you're doing a lot of merging or splitting of lists. This is because they can just mangle some pointers instead of actually copying the data. Even if you're doing a lot of insertion in the middle of the list, RingBuf can still be better because of how expensive it is to seek to the middle of a LinkedList.

**Comment by Manishearth** [comment]
Thanks, that looks good to me. I'll start a wiki later today.

**Comment by llogiq** [comment]
@Manishearth Now ten days later, allow me the question: Where did you want to start that wiki?

**Comment by Manishearth** [comment]
hah

https://github.com/Manishearth/rust-clippy/wiki

**Comment by llogiq** [comment]
I just started the wiki. Perhaps we may want to link the respective pages from the README (@birkenfeld)?

**Comment by Manishearth** [comment]
Perhaps we should all keep them in a single page with section links so that it's easy to autogenerate links?

(Alternatively, one page per lint)

**Comment by llogiq** [comment]
I've started out with one page per lint (or two lints, as in the `*_bit_mask` case), but I'm open to changing it to one page.

**Comment by Manishearth** [comment]
one page per lint is fine, but one page for two lints becomes hard to auto-link to :)

**Comment by llogiq** [comment]
Another reason for putting all on one page.

**Comment by Manishearth** [comment]
Meh, it works out equally if you have one-lint-per-appropriately-named-page

**Comment by llogiq** [comment]
I think the wiki now has ample information on all the lints, and we can improve it as we go.
